streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -439,7 +440,20 @@ public Subscription subscription(final Set<String> topics) {
                                     } else {
                                         numPartitionsCandidate = metadata.partitionCountForTopic(sourceTopicName);
                                         if (numPartitionsCandidate == null) {
-                                            repartitionTopicMetadata.get(topicName).numPartitions = NOT_AVAILABLE;
+                                          assignment = new HashMap<>();
+                                          for (final ClientMetadata clientMetadata : clientsMetadata.values()) {
+                                            for (final String consumerId : clientMetadata.consumers) {
+                                              assignment.put(consumerId, new Assignment(
+                                                  Collections.emptyList(),
+                                                  new AssignmentInfo(AssignmentInfo.LATEST_SUPPORTED_VERSION,
+                                                      Collections.emptyList(),
+                                                      Collections.emptyMap(),
+                                                      Collections.emptyMap(),
+                                                      AssignmentInfo.UNKNOWN_PARTITION).encode()
+                                              ));
+                                            }
+                                          }
+                                          return assignment;
```

Comment:
As mentioned in the JIRA ticket, we should log an ERROR that `some of the source topics ( + source topic lists) are not known yet during rebalance, please make sure they have been pre-created before starting the Streams application.`